core/models/guild.js

```diff
@@ -24,10 +24,18 @@ class Guild extends GuildModel {
         global.discotron.on("plugin-deleted", (pluginId) => {
             this.onPluginDeleted(pluginId);
         });
+    }
 
-        this._init();
-
-        Guild._guilds[discordId] = this;
+    /**
+     * Creates a new guild which is initialized and registered.
+     * @param {string} discordId Discord guild id
+     * @returns {Guild} New guild instance.
+     */
+    static create(discordId) {
+        const guild = new Guild(discordId);
+        guild._init();
```

Comment:
Maybe we should make `init` public to stay consistant with what was done in plugin.js? Also we could consider calling that in the constructor if init should always be called after creating the object

Reply:
> make public

Agreed.

> call in constructor

`init` has a database insert and many selects, those should not be run in constructors.

Reply:
BTW what about `async` create/init functions? in this case, guild.init is async - should we await the call here and make create async too?